controllers/infra/drivers/aws/create_instance.go

```diff
@@ -129,7 +142,15 @@ func (d Driver) createInstances(hosts *v1.Hosts, infra *v1.Infra) error {
 			Value: &value,
 		},
 	)
+	if infra.Spec.SSH.PkName == "" {
+		if err := d.createKeyPair(infra); err != nil {
+			return err
+		}
+	}
 	keyName := infra.Spec.SSH.PkName
+	//todo use ami to search root device name
+	rootDeviceName := "/dev/xvda"
```

Comment:
Seems we can add this config data in `infra.yaml`~

Reply:
rootDeviceName associated to ami.
infra.yaml already had ami(image).
todo: use ami to search root device name by calling API or self-built map.

Reply:
no necessary change in yaml